Gemfile

```diff
@@ -7,3 +7,9 @@ rails_version = "#{ENV['RAILS_VERSION'] || 'main'}"
 
 gem "capybara", "~> 3"
 gem "rails", rails_version == "main" ? { git: "https://github.com/rails/rails", ref: "main" } : rails_version
+
+if RUBY_VERSION >= "3.1"
+  gem "net-imap", require: false
```

Comment:
What happens if we omit these gems?

Reply:
So this is a known Rails bug, fixed on the Rails 7 branch (with Rails 7.0.1) but not on Rails 6.1.  You can find a discussion [here](https://github.com/mikel/mail/pull/1439#issuecomment-1002801221)

These gems were included as defaults in Ruby versions 3.0 and earlier, but are not included in 3.1+.  Rails depends on them transitively through the mail gem.  [Rails 7.0.1 fixed this](https://github.com/rails/rails/pull/44083) but it can't be fixed by Rails for 6.x because of the need to support Ruby 2.6, where this inclusion causes an error.

On a more practical note, if these are left out then Ruby 3.1 / Rails 6.1 builds fail with an error saying they can't find net-smtp.